app/src/main/java/com/example/newsapp/ui/screen/newslist/view/NewsListFragment.kt

```diff
@@ -0,0 +1,103 @@
+package com.example.newsapp.ui.screen.newslist.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.viewModels
+import androidx.navigation.Navigation
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.newsapp.R
+import com.example.newsapp.data.components.network.repository.news.model.Article
+import com.example.newsapp.databinding.FragmentNewsListBinding
+import com.example.newsapp.ui.base.BaseFragment
+import com.example.newsapp.ui.components.adapters.base.BaseListAdapter
+import com.example.newsapp.ui.components.listeners.ItemOnClickListener
+import com.example.newsapp.ui.screen.newslist.viewmodel.NewsListFragmentViewModel
+import com.example.newsapp.util.SectionItemBuilder
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class NewsListFragment : BaseFragment<NewsListFragmentViewModel, FragmentNewsListBinding>(),
+    ItemOnClickListener {
+
+    private fun initControls() {
+
+    }
+
+    private fun initialize() {
+        initControls()
+        initData()
+
+        if(viewModel.newsResponse.value == null) {
```

Comment:
?.let, ?.run, ?.also, ?apply ve with() bunlara çalışıp blog post atar mısın? önceliğin bu olsun elindeki yazılar bitince